scripts/add_gen3_repo.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python
+
+"""Convert a Gen 2 dataset to a Gen 3 dataset.
+
+By default, this creates a hybrid Gen 2/3 dataset with shared files. A flag
+lets a dataset be permanently migrated to Gen 3 instead.
+"""
+
+import argparse
+import os
+import shutil
+import tempfile
+
+import lsst.log
+import lsst.daf.butler as daf_butler
+from lsst.obs.base.script import convertGen2RepoToGen3
+import lsst.ap.verify as ap_verify
+
+
+class _Parser(argparse.ArgumentParser):
+    def __init__(self, **kwargs):
+        # super() causes problems with program name
+        argparse.ArgumentParser.__init__(
+            self,
+            description="Copy a dataset's Gen 2 files into the Gen 3 format, overwriting a previous copy "
+                        "if necessary. This creates a hybrid Gen 2/3 dataset unless the --drop-gen2 flag "
+                        "is provided. DO NOT delete the Gen 2 files unless this flag has been used, as the "
+                        "Gen 3 part of a hybrid dataset depends on them.\n\n"
+                        "Assumes that the dataset's config directory has three configs for "
+                        "obs.base.gen2to3.ConvertRepoTask: convertRepo_calibs.py, convertRepo_copied.py and "
+                        "convertRepo_templates.py. See ap_verify_dataset_template/config for examples.",
+            **kwargs)
+        self.add_argument("--dataset", required=True,
+                          help="The name of the dataset as recognized by ap_verify.py.")
+        self.add_argument("--drop-gen2", action="store_true",
+                          help="Create a standalone Gen 3 repo instead of sharing files with Gen 2. "
+                               "Intended for use only once ap_verify no longer supports Gen 2.")
```

Comment:
Why is this a subclass of `ArgumentParser` rather than a factory function?

Reply:
Because `argparse` is an object-oriented CLI library, and most such libraries are designed on the assumption that they will be subclassed by the application. I haven't seen any Python documentation specifically saying to do it that way, but it's basically my default.